### PR TITLE
Feature/issue/223

### PR DESCRIPTION
**Pull request by vbence86**
### Intent
To add functionality to identify CORS exception while attempting to redirect the browser to an external URL and trigger a traditional ```location.href``` redirect.

### Dependency
https://github.com/metal/metal-ajax/pull/8

### CODE REVIEW

**Comment by eduardolundgren** [comment]
This should not be necessary, Senna doesn't navigate to external URLs already.

**Comment by jbalsas** [comment]
Hey @eduardolundgren, this is meant to address the CORS redirect issue together with https://github.com/metal/metal-ajax/pull/8

File: src/app/App.js [hunk 1]
```diff
@@ -568,6 +568,9 @@ class App extends EventEmitter {
 	 */
 	handleNavigateError_(path, nextScreen, err) {
 		console.log('Navigation error for [' + nextScreen + '] (' + err + ')');
+		if (err.isBlockedByCORS) {
```
Comment (eduardolundgren) [inline src/app/App.js]: We don't route external domains, therefore CORS will never happen, although this error is to handle masked CORS that comes from a redirect from a URL of the same domain of the page. We can prob here rename to something more specific, e.g. `isBlockedByCORSAfterRedirect`.

**Comment by fernandosouza** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by jbalsas** [comment]
As we've been discussing, the fallback should happen on the App level, since based on the business logic, we might want to do different things. We probably want to emit an event so the app developer can run its custom logic.

Also, see https://github.com/metal/metal-ajax/pull/8#issuecomment-318630180 for some changes in the approach we could take to identify the issue.

**Pull request closed without merging**